api-parent/core-api/src/main/java/com/sap/cloud/environment/servicebinding/api/DefaultServiceBinding.java

```diff
@@ -100,6 +104,17 @@ public Optional<String> getServiceName()
         return Optional.ofNullable(serviceName);
     }
 
+    @Nonnull
+    @Override
+    public Optional<ServiceIdentifier> getServiceIdentifier()
+    {
+        if( serviceIdentifier != null ) {
+            return Optional.of(serviceIdentifier);
+        }
+
+        return getServiceName().map(ServiceIdentifier::of);
```

Comment:
Question: If we are able to derive a value here, doesn't it make sense to also save that to the `serviceIdentifier` member variable?